hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/TransactionMetadataBuilder.java

```diff
@@ -112,6 +133,22 @@ public T txnBody(@NonNull TransactionBody txn) {
         return self();
     }
 
+    /**
+     * Adds a {@link Set} of read keys for a {@link com.hedera.node.app.spi.state.ReadableKVState}
+     *
+     * @param statesKey the key of the {@link com.hedera.node.app.spi.state.ReadableStates}
+     * @param stateKey the key of the {@link com.hedera.node.app.spi.state.ReadableKVState}
+     * @param readKeys the read keys
+     * @return builder object
+     */
+    public T addReadKeys(
+            final String statesKey,
+            final String stateKey,
+            final Set<? extends Comparable<?>> readKeys) {
+        this.readKeys.add(new TransactionMetadata.ReadKeys(statesKey, stateKey, readKeys));
```

Comment:
If the ReadKeys record is private API we can add a `addReadKeys(ReadKeys keys)` method.